sympy/functions/special/spherical_harmonics.py

```diff
@@ -189,16 +192,22 @@ def _eval_rewrite_as_sin(self, n, m, theta, phi, **kwargs):
     def _eval_rewrite_as_cos(self, n, m, theta, phi, **kwargs):
         # This method can be expensive due to extensive use of simplification!
         from sympy.simplify import simplify, trigsimp
-        # TODO: Make sure n \in N
-        # TODO: Assert |m| <= n ortherwise we should return 0
-        term = simplify(self.expand(func=True))
+        if all(map(lambda x: x.is_integer, (n, m, theta, phi))):
+            if n in S.Naturals and Abs(m) <= n:
+                term = simplify(self.expand(func=True))
+            elif Abs(m) < n:
+                term = S.Zero
+        else:
+            term = simplify(self.expand(func=True))
         # We can do this because of the range of theta
         term = term.xreplace({Abs(sin(theta)):sin(theta)})
         return simplify(trigsimp(term))
 
     def _eval_conjugate(self):
-        # TODO: Make sure theta \in R and phi \in R
         n, m, theta, phi = self.args
+        if all(map(lambda x: x.is_integer, (n, m, theta, phi))):
+            if theta in S.Reals and phi in S.Reals:
+                return S.NegativeOne**m * self.func(n, -m, theta, phi)
```

Comment:
This seems like it doesn't add anything as you do the same thing independent of the if-cases.

Reply:
Also, I do not think theta and phi should be integer?

Reply:
Sorry, by mistakenly added the wrong code.  I'll commit the correct code again.

Reply:
> Also, I do not think theta and phi should be integer?

I think the check should be is_Number but I'm not sure.